lib/rex/proto/http/client.rb

```diff
@@ -600,8 +600,8 @@ def read_response(t = -1, opts = {})
           rblob = rbody.to_s + rbufq.to_s
           tries = 0
           begin
-            # XXX: This doesn't deal with chunked encoding or "Content-type: text/html; charset=..."
-            while tries < 1000 and resp.headers["Content-Type"]== "text/html" and rblob !~ /<\/html>/i
+            # XXX: This doesn't deal with chunked encoding
+            while tries < 1000 and resp.headers["Content-Type"] and resp.headers["Content-Type"].starts_with?('text/html') and rblob !~ /<\/html>/i
```

Comment:
Normally, I'm squishy on changing core lib code, but this looks fine.  (Famous last words)  The code is 12 years old and looks to be designed to handle an edge case.  It appears functionally equivalent for the previous instances, and I'm curious how often this loop has to run in most cases?